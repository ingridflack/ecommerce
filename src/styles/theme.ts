const theme = {
  colors: {
    background: "#fbfefd",

    gray: {
      white: "#FFFFFF",
      platinum: "#e1e1e6",
      silver: "#a8a8b3",
      dark: "#141414",
    },

    primary: "#52b591",
    primaryDark: "#49a282",
  },
  fontSize: {
    small: "0.75rem",
    medium: "0.875rem",
    normal: "1rem",
    large: "1.5rem",
  },
  spacings: {
    tiny: "0.4rem",
    small: "0.8rem",
    medium: "1rem",
    big: "2rem",
  },
} as const;

export type Theme = typeof theme;

export default theme;
