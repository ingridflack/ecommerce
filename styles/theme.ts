const theme = {
  colors: {
    background: "#fbfefd",

    gray: {
      white: "#FFFFFF",
      platinum: "#e1e1e6",
      silver: "#a8a8b3",
      dark: "#141414",
    },

    primary: "#52b591",
  },
  fontSize: {
    small: "0.75rem",
    medium: "0.875rem",
    normal: "1rem",
    large: "1.5rem",
  },
} as const;

export type Theme = typeof theme;

export default theme;
