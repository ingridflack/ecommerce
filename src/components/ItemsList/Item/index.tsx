import {
  MdAddCircleOutline,
  MdDelete,
  MdRemoveCircleOutline,
} from "react-icons/md";
import { IProductItem } from "../../../interfaces";

import * as S from "../styles";

interface ItemProps {
  product: any;
  onProductDecrement: (product: IProductItem) => void;
  onProductIncrement: (product: IProductItem) => void;
  onRemoveProduct: (productId: string) => void;
}

export const Item = ({
  product,
  onProductDecrement,
  onProductIncrement,
  onRemoveProduct,
}: ItemProps) => {
  return (
    <tr>
      <S.TableColumn>
        <strong>{product.name}</strong>
      </S.TableColumn>
      <S.TableColumn>
        <S.Quantity>
          <button
            type="button"
            data-testid="decrement-btn"
            disabled={product.quantity <= 1}
            onClick={() => onProductDecrement(product)}
          >
            <MdRemoveCircleOutline size={20} />
          </button>
          <input readOnly value={product.quantity} />
          <button
            type="button"
            data-testid="increment-btn"
            onClick={() => onProductIncrement(product)}
          >
            <MdAddCircleOutline size={20} />
          </button>
        </S.Quantity>
      </S.TableColumn>
      <S.TableColumn>
        <strong>{product.subTotal}</strong>
      </S.TableColumn>
      <S.TableColumn>
        <button
          type="button"
          data-testid="remove-btn"
          onClick={() => onRemoveProduct(product.id)}
        >
          <MdDelete size={20} />
        </button>
      </S.TableColumn>
    </tr>
  );
};

export default Item;
