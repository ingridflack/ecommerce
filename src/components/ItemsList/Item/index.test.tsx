import { fireEvent, render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../../../styles/theme";
import Item from ".";

jest.mock("../../../hooks/useCart");

const renderTableContainer = (children: any) => (
  <ThemeProvider theme={theme}>
    <table>
      <tbody>{children}</tbody>
    </table>
  </ThemeProvider>
);

describe("Item", () => {
  const mockedProduct = {
    quantity: 2,
    id: 1,
    price: 179.9,
    title: "Produto teste",
  };

  const handleProductDecrementMock = jest.fn();
  const handleProductIncrementMock = jest.fn();
  const handleRemoveProductMock = jest.fn();

  it("should match the snapshot", () => {
    const { container } = render(
      renderTableContainer(
        <Item
          product={mockedProduct}
          onProductDecrement={handleProductDecrementMock}
          onProductIncrement={handleProductIncrementMock}
          onRemoveProduct={handleRemoveProductMock}
        />
      )
    );

    expect(container).toMatchSnapshot();
  });

  it("should call handleProductDecrement function", () => {
    const { getByTestId } = render(
      renderTableContainer(
        <Item
          product={mockedProduct}
          onProductDecrement={handleProductDecrementMock}
          onProductIncrement={handleProductIncrementMock}
          onRemoveProduct={handleRemoveProductMock}
        />
      )
    );

    const productDecrementBtn = getByTestId("decrement-btn");
    fireEvent.click(productDecrementBtn);

    expect(handleProductDecrementMock).toHaveBeenCalledWith(mockedProduct);
  });

  it("should call handleProductIncrement function", () => {
    const { getByTestId } = render(
      renderTableContainer(
        <Item
          product={mockedProduct}
          onProductDecrement={handleProductDecrementMock}
          onProductIncrement={handleProductIncrementMock}
          onRemoveProduct={handleRemoveProductMock}
        />
      )
    );

    const productDecrementBtn = getByTestId("decrement-btn");
    fireEvent.click(productDecrementBtn);

    expect(handleProductDecrementMock).toHaveBeenCalledWith(mockedProduct);
  });

  it("should call handleRemoveProduct function", () => {
    const { getByTestId } = render(
      renderTableContainer(
        <Item
          product={mockedProduct}
          onProductDecrement={handleProductDecrementMock}
          onProductIncrement={handleProductIncrementMock}
          onRemoveProduct={handleRemoveProductMock}
        />
      )
    );

    const productDecrementBtn = getByTestId("decrement-btn");
    fireEvent.click(productDecrementBtn);

    expect(handleProductDecrementMock).toHaveBeenCalledWith(mockedProduct);
  });
});
