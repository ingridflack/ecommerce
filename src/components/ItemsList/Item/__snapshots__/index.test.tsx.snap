// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match the snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="sc-9204a22b-1 caQlDm"
        >
          <strong />
        </td>
        <td
          class="sc-9204a22b-1 caQlDm"
        >
          <div
            class="sc-9204a22b-2 cfnsnr"
          >
            <button
              data-testid="decrement-btn"
              type="button"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </button>
            <input
              readonly=""
              value="2"
            />
            <button
              data-testid="increment-btn"
              type="button"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </button>
          </div>
        </td>
        <td
          class="sc-9204a22b-1 caQlDm"
        >
          <strong />
        </td>
        <td
          class="sc-9204a22b-1 caQlDm"
        >
          <button
            data-testid="remove-btn"
            type="button"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
